controllers/topology/cluster_controller_test.go

```diff
@@ -283,51 +286,143 @@ func TestClusterReconciler_reconcileUpdatesOnClusterClass(t *testing.T) {
 		WithSpecFields(map[string]interface{}{"spec.template.spec.fakeSetting": true}).
 		Build()
 
-	infrastructureClusterTemplate := builder.InfrastructureClusterTemplate(ns.Name, "infraclustertemplate").
-		// Create spec fake setting to assert that template spec is non-empty for tests.
-		WithSpecFields(map[string]interface{}{"spec.template.spec.fakeSetting": true}).
+	infrastructureClusterTemplate1 := builder.InfrastructureClusterTemplate(ns.Name, "infraclustertemplate").
+		Build()
+
+	infrastructureClusterTemplate2 := builder.InfrastructureClusterTemplate(ns.Name, "infraclustertemplate2").
 		Build()
 	controlPlaneTemplate := builder.ControlPlaneTemplate(ns.Name, "cp1").
-		WithInfrastructureMachineTemplate(infrastructureMachineTemplate).
+		WithInfrastructureMachineTemplate(infrastructureMachineTemplate1).
 		Build()
 	bootstrapTemplate := builder.BootstrapTemplate(ns.Name, "bootstraptemplate").Build()
 
 	// 2) ClusterClass definitions including definitions of MachineDeploymentClasses used inside the ClusterClass.
 	machineDeploymentClass1 := builder.MachineDeploymentClass(workerClassName1).
-		WithInfrastructureTemplate(infrastructureMachineTemplate).
+		WithInfrastructureTemplate(infrastructureMachineTemplate1).
 		WithBootstrapTemplate(bootstrapTemplate).
 		WithLabels(map[string]string{"foo": "bar"}).
 		WithAnnotations(map[string]string{"foo": "bar"}).
 		Build()
 	machineDeploymentClass2 := builder.MachineDeploymentClass(workerClassName2).
-		WithInfrastructureTemplate(infrastructureMachineTemplate).
+		WithInfrastructureTemplate(infrastructureMachineTemplate1).
 		WithBootstrapTemplate(bootstrapTemplate).
 		Build()
-	clusterClass := builder.ClusterClass(ns.Name, clusterClassName).
-		WithInfrastructureClusterTemplate(infrastructureClusterTemplate).
+	machineDeploymentClass3 := builder.MachineDeploymentClass(workerClassName3).
+		WithInfrastructureTemplate(infrastructureMachineTemplate2).
+		WithBootstrapTemplate(bootstrapTemplate).
+		Build()
+	clusterClass := builder.ClusterClass(ns.Name, clusterClassName1).
+		WithInfrastructureClusterTemplate(infrastructureClusterTemplate1).
 		WithControlPlaneTemplate(controlPlaneTemplate).
-		WithControlPlaneInfrastructureMachineTemplate(infrastructureMachineTemplate).
+		WithControlPlaneInfrastructureMachineTemplate(infrastructureMachineTemplate1).
 		WithWorkerMachineDeploymentClasses(*machineDeploymentClass1, *machineDeploymentClass2).
 		Build()
 
+	// This ClusterClass changes a number of things in a ClusterClass in a way that is compatible for a ClusterClass rebase operation.
+	// 1) It changes the controlPlaneMachineInfrastructureTemplate to a new template.
+	// 2) It adds a new machineDeploymentClass
+	// 3) It changes the infrastructureClusterTemplate.
+	clusterClassForRebase := builder.ClusterClass(ns.Name, clusterClassName2).
+		// TODO: Is changing the infrastructureClusterTemplate an allowed operation?
```

Comment:
Yes, it is the recommended way to update InfrastructureClusterTemplates. It's a similar case to ControlPlaneTemplate. It will lead to changes in the same InfrastructureCluster though (and no rotation)